### PR TITLE
Use shared dependency for attribution metadata writing

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes

The latest version of the `@aws-amplify/backend-output-storage` package exposes functionality for storing BI metrics in the stack description. This PR replaces the logic in the data construct to set this metadata with a call to the new library method.

The new version of the library is also CJS compliant, so I have removed all of the CJS building gymnastics from the build command of the data construct.

#### Description of how you validated changes
Existing tests pass and added a unit test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [x] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by alharris-at — DISMISSED** [review]

**Comment by alharris-at** [comment]
Failure is a yarn.lock mismatch, but that's been failing consistently today. We're going to hold off on merging until after the feature branches are merged in.

**Review by dpilch — DISMISSED** [review]

**Review by alharris-at — APPROVED** [review]

**Review by dpilch — APPROVED** [review]

**Pull request merged**
